modules/virtual_network_gateway/main.tf

```diff
@@ -144,38 +147,38 @@ resource "azurerm_local_network_gateway" "this" {
 resource "azurerm_virtual_network_gateway_connection" "this" {
   for_each = var.local_network_gateways
 
-  name                = "${var.name_prefix}con-vgw${var.name_suffix}-${each.value.connection}"
+  name                = each.value.connection_name
   location            = var.location
   resource_group_name = var.resource_group_name
 
-  type                       = "IPsec"
+  type                       = var.connection_type
   virtual_network_gateway_id = azurerm_virtual_network_gateway.this.id
   local_network_gateway_id   = azurerm_local_network_gateway.this[each.key].id
 
-  enable_bgp                     = try(var.enable_bgp, false)
-  local_azure_ip_address_enabled = try(var.private_ip_address_enabled, true)
+  enable_bgp                     = var.enable_bgp
+  local_azure_ip_address_enabled = var.private_ip_address_enabled
   shared_key                     = var.ipsec_shared_key
 
   dynamic "custom_bgp_addresses" {
-    for_each = try(each.value.custom_bgp_addresses, {})
+    for_each = each.value.custom_bgp_addresses
```

Comment:
this is defined as an optional list, w/o any default value, hence `null` - this will fail -> add a default value of `[]` to type

Reply:
added